plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,290 @@
+func (l *Logstash) createHTTPClient() (*http.Client, error) {
+
+	tr := &http.Transport{
+		ResponseHeaderTimeout: time.Duration(3 * time.Second),
+	}
+
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   time.Duration(4 * time.Second),
+	}
+
+	return client, nil
+}
+
+func (l *Logstash) gatherJSONData(url string, v interface{}) error {
+
+	r, err := l.client.Get(url)
+	if err != nil {
+		return err
+	}
+
+	defer r.Body.Close()
+
+	if r.StatusCode != http.StatusOK {
+		return fmt.Errorf("Logstash: API responded with status-code %d, expected %d",
+			r.StatusCode, http.StatusOK)
+	}
+
+	if err = json.NewDecoder(r.Body).Decode(v); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *Logstash) gatherJVMStats(url string, acc telegraf.Accumulator) error {
+	JVMStats := &JVMStats{}
+	if err := l.gatherJSONData(url, JVMStats); err != nil {
+		return err
+	}
+
+	tags := map[string]string{
+		"node_id": JVMStats.CommonData.ID,
+	}
+
+	stats := map[string]interface{}{
+		"jvm": JVMStats.JVM,
+	}
+
+	now := time.Now()
+	for p, s := range stats {
+		f := jsonparser.JSONFlattener{}
+		// parse Json, ignoring strings and bools
+		err := f.FlattenJSON("", s)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("logstash_"+p, f.Fields, tags, now)
+	}
+
+	return nil
+}
+
+func (l *Logstash) gatherProcessStats(url string, acc telegraf.Accumulator) error {
+	ProcessStats := &ProcessStats{}
+	if err := l.gatherJSONData(url, ProcessStats); err != nil {
+		return err
+	}
+
+	tags := map[string]string{
+		"node_id": ProcessStats.CommonData.ID,
+	}
+
+	stats := map[string]interface{}{
+		"process": ProcessStats.Process,
+	}
+
+	now := time.Now()
+	for p, s := range stats {
+		f := jsonparser.JSONFlattener{}
+		// parse Json, ignoring strings and bools
+		err := f.FlattenJSON("", s)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("logstash_"+p, f.Fields, tags, now)
+	}
+
+	return nil
+}
+
+func (l *Logstash) gatherPipelineStats(url string, acc telegraf.Accumulator) error {
+	PipelineStats := &PipelineStats{}
+	if err := l.gatherJSONData(url, PipelineStats); err != nil {
+		return err
+	}
+
+	tags := map[string]string{
+		"node_id": PipelineStats.CommonData.ID,
+	}
+
+	stats := map[string]interface{}{
+		"events": PipelineStats.Pipeline.Events,
+	}
+
+	now := time.Now()
+	// Events
+	for p, s := range stats {
+		f := jsonparser.JSONFlattener{}
+		// parse Json, ignoring strings and bools
+		err := f.FlattenJSON("", s)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("logstash_"+p, f.Fields, tags, now)
+	}
+
+	// Input Plugins
+	for _, plugin := range PipelineStats.Pipeline.Plugins.Inputs {
+		//plugin := &plugin
+		fields := map[string]interface{}{
+			"name":               plugin.Name,
+			"duration_in_millis": plugin.Events.DurationInMillis,
+			"in":                 plugin.Events.In,
+			"out":                plugin.Events.Out,
+		}
+		acc.AddFields("logstash_plugin_input_"+plugin.Name, fields, tags, now)
```

Comment:
For each of the plugins, I suggest this structure:
```
logstash_plugins,plugin=<name>type=input in=42 out=42, duration_in_millis=42
logstash_plugins,plugin=<name>type=output in=42 out=42, duration_in_millis=42
logstash_plugins,plugin=<name>type=filter in=42 out=42, duration_in_millis=42
```

It also looks like we should add support for `queue_push_duration_in_milli` on inputs and logstash_events.

Reply:
Hmm, I tried this way but looks like if i will use comma - ',' in name, it will be escaped.
With https://github.com/lkmcs/telegraf/blob/master/plugins/inputs/logstash/logstash.go#L226 I will get `logstash_plugins\,plugin=stdout\,type=output,node_id=e3fc20c4-d717-417f-88ac-417a56d4c39c,host=laptop in=0,out=0,name="stdout",duration_in_millis=0 1510868747000000000`

Reply:
I need to use tags, not name, got it :)